lib/streamlit/__init__.py

```diff
@@ -311,7 +316,7 @@ def experimental_show(*args):
         exception(exc)
 
 
-def experimental_get_query_params():
+def experimental_get_query_params() -> Union[Dict[str, List[str]], str]:
```

Comment:
Hm, I think it'd be preferable to change this return type to `Dict[str, List[str]]` and the return value in the `ctx is None` case to be an empty dict.

The current return value in this case feels like it's just an error to me, and while changing it is technically a breaking change, I think it's still fine to do as

1. the `ctx is None` case is only hit when users are running streamlit "incorrectly" anyway (`python3 my_script.py` vs `streamlit run my_script.py`)
2. the function is in the experimental namespace, so we're not committed to very strictly following SemVer for it

Reply:
Was thinking the same thing, was just trying to avoid change of behaviour. My goal is really just to get "descriptive" typing in place. As can be seen here, that can serve as a really good starting point for discovering needless complexity in the future. Just typing out the types explicitly make them easier to discuss.

But, I don't mind updating this function, as you suggest. So with your blessing, I've done that.